deleteAllProgramsData.sh

```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-if [ -z "$2" ]; then
+if [ -z "$1" ] || [ -z "$2" ]; then
```

Comment:
I think condition "[ -z "$1" ]" is redundant because if we have not two parameters then always condition is true. If you want to improve the readability you can change the condition to "[ $# -lt 2 ]".

Reply:
Ok